src/mount/zwo.rs

```diff
@@ -0,0 +1,283 @@
+
+        SlewSpeed::Specific(s) => {
+            let is_close = |req: f64, actual: f64| { let rel = req.abs() / actual; rel >= 0.99 && rel <= 1.01 };
+
+            for sss in SUPPORTED_SLEWING_SPEEDS.iter().take(SUPPORTED_SLEWING_SPEEDS.len() - 1) {
+                if is_close(s.0, sss.speed.0) { return Some(sss); }
+            }
+
+            None
+        }
+    }
+}
+
+impl ZWO {
+    #[must_use]
+    pub fn new(device: &str) -> Result<ZWO, Box<dyn Error>> {
+        let mut serial_port = serialport::new(device, 9600)
+            .data_bits(serialport::DataBits::Eight)
+            .flow_control(serialport::FlowControl::None)
+            .parity(serialport::Parity::None)
+            .stop_bits(serialport::StopBits::One)
+            .timeout(std::time::Duration::from_millis(50))
+            .open()?;
+
+
+        let model = if let Ok(chars) = send_cmd_and_get_reply(
+            &mut serial_port,
+            ":GVP#".into(),
+            ResponseType::EndsWith(END_CHAR)
+        ) {
+            if let Ok(s) = String::from_utf8(chars) {
+                match s.as_str() {
+                    "AM3" => "AM3".into(),
+                    "AM5" => "AM5".into(),
+                    _ => format!("(unknown - {})", s)
+                }
+            } else {
+                "(unknown) string".into()
+            }
+        } else {
+            "(unknown) command error".into()
+        };
+
+        // Disable tracking
+        send_cmd_and_get_reply(&mut serial_port, ":Td#".into(), ResponseType::None).map(|_| ())?;
+
+        Ok(ZWO{
+            model: model,
+            device: device.to_string(),
+            tracking: false,
+            serial_port
+        })
+    }
+}
+
+impl Drop for ZWO {
+    fn drop(&mut self) {
+        let _ = self.stop();
+    }
+}
+
+impl Mount for ZWO {
+    fn get_info(&self) -> String {
+        format!("ZWO {} on {}", self.model, self.device)
+    }
+
+
+    fn set_tracking(&mut self, enabled: bool) -> Result<(), Box<dyn Error>> {
+        match send_cmd_and_get_reply(
+            &mut self.serial_port,
+            format!(":T{}#",  if enabled { "e" } else { "d" }),
+            ResponseType::CharsReceived("1".into())
+        ) {
+            Ok(_) => { self.tracking = enabled; Ok(()) },
+            Err(e) => Err(e)
+        }
+    }
+
+    fn guide(&mut self, axis1_speed: RadPerSec, axis2_speed: RadPerSec) -> Result<(), Box<dyn Error>> {
+        if !(axis1_speed.is_zero() && axis2_speed.is_zero()) && !self.tracking {
+            return Err("cannot guide when tracking is disabled".into());
+        }
+
+        const MIN_SIDEREAL_MULT: f64 = 0.1;
+
+        let a1_s = (axis1_speed.0.abs() / SIDEREAL_RATE.0).max(MIN_SIDEREAL_MULT);
+        let a2_s = (axis2_speed.0.abs() / SIDEREAL_RATE.0).max(MIN_SIDEREAL_MULT);
+
+        if axis1_speed.is_zero() {
+            send_cmd_and_get_reply(&mut self.serial_port, ":Mgn0000#".into(), ResponseType::None).map(|_| ())?;
```

Comment:
It seems it should be `e`, `w` in this block (axis1 = RA) and `n`, `s` in the next block (axis2 = Dec),  similarly in blocks below.

Reply:
You are right, thanks